src/toil_scripts/gatk_pipeline/align-and-call

```diff
@@ -0,0 +1,260 @@
+    This function should be unpacked for every item being returned, unless none
+    of the paths are needed in which case it should be unassigned.
+
+    work_dir: str       Path to the current working directory
+    ids: dict           Dictionary of fileStore IDs, accessed by filename (str)
+    *args: str(s)       Files to be retrieved and placed in the current working directory
+    """
+    paths = OrderedDict()
+    for name in args:
+        if not os.path.exists(os.path.join(work_dir, name)):
+            file_path = job.fileStore.readGlobalFile(ids[name], os.path.join(work_dir, name))
+        else:
+            file_path = os.path.join(work_dir, name)
+        paths[name] = file_path
+        if len(args) == 1:
+            return file_path
+    return paths.values()
+
+
+def docker_call(work_dir, tool_parameters, tool, java_opts=None,
+                outfiles=None, sudo=False):
+    """
+    Makes subprocess call of a command to a docker container.
+    """
+    base_docker_call = 'docker run --log-driver=none --rm -v {}:/data'.format(work_dir).split()
+    if sudo:
+        base_docker_call = ['sudo'] + base_docker_call
+    if java_opts:
+        base_docker_call = base_docker_call + ['-e', 'JAVA_OPTS={}'.format(java_opts)]
+    if debug:
+        for outfile in outfiles:
+            outpath = os.path.join(work_dir, outfile)
+            f = open(outpath, 'w')
+            f.write('debug')
+            f.close()
+        return
+    try:
+        subprocess.check_call(base_docker_call + [tool] + tool_parameters)
+    except subprocess.CalledProcessError, e:
+        raise RuntimeError('docker command returned a non-zero exit status. {}'.format(e))
+    except OSError:
+        raise RuntimeError('docker not found on system. Install on all nodes.')
+    for outfile in outfiles:
+        assert os.path.exists(os.path.join(work_dir, outfile))
+
+
+# START - >
+def download_shared_files(job, input_args):
+    """
+    Downloads shared files that are used by all samples for alignment and places them in the jobstore.
+
+    input_args: dict        Input arguments (passed from main())
+    """
+    shared_files = ['ref.fa', 'phase.vcf', 'mills.vcf', 'dbsnp.vcf',
+                    'omni.vcf', 'hapmap.vcf', 'mills.vcf']
+    shared_ids = {}
+    for fname in shared_files:
+        url = input_args[fname]
+        shared_ids[fname] = job.addChildJobFn(download_from_url, url,
+                                              fname).rv()
+    job.addFollowOnJobFn(make_index, shared_ids, input_args)
+
+
+def make_index(job, shared_ids, input_args):
+    """
+    Makes necessary reference indexes for BWA alignment
+    """
+    work_dir = job.fileStore.getLocalTempDir()
+    sudo = input_args['sudo']
+    # Retrieve samples
+    return_input_paths(job, work_dir, shared_ids, 'ref.fa')
+    # Call: BWA
+    parameters = ["index", "ref.fa"]
+    docker_call(tool='quay.io/ucsc_cgl/bwa:0.7.12--dd5ac549b95eb3e5d166a5e310417ef13651994e',
```

Comment:
this step can take an awful long time, #73 instead requires the bwa indices as input parameters